### PR TITLE
adds section 31 agent, custom objectives not made yet.

### PR DESCRIPTION
**Pull request by 2cwldys**
[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

I thought adding section 31 would be very interesting, they are now a thing. The Free Objective does not mean murderbone, however it means "Defend the best interests and survival of the federation, at all costs. Even if it includes espionage and acts of rank deception within Starfleet."

### CODE REVIEW

**Comment by 2cwldys** [comment]
I tested all of these and they now work without issues, it was a huge learning experience. I PM'd like over ten people or close to it?

**Comment by 2cwldys** [comment]
The player starts with the following:

- Their newly found objectives, they may even involve incapacitating a crewmember.
- Phaser in the Right Hand.
- Nightvision Goggles in your bag.
- Agent ID Forging Kit (Chameleon ID)
- Hypospray with Chloral Hydrate bottle, to blur and almost cause sleep inducing pain to others, so that you can wake up infront of them in your black leather turtleneck.
- Section 31 Agent's Tactical Turtleneck
- Cyanide to kill yourself before you are captured, so you don't have to use the explosive implant.
- A "Captain's Spare" to swipe with your "Agent ID Forging Kit" (Chameleon ID)
- Freedom (4 uses), Explosive, and Mindshield Implants; imbedded by the finest of Section 31 Physicians and Surgeons.
- A telescopic baton, to have a nonlethal approach to getting others out of your way.
- A PDA in the other pocket, if needed be; best for holding the "Agent ID Forging Kit." (Chameleon ID)
- A theoretical and literal license to kill for the best interests and survival of the federation (objectives and rp escalation), but not to murderbone.
- Your orders are your own, aside from the Objectives; essentially. Clandestine Ops.
- Potentially up to two other agents, depending on the size and population of current vessel.
- A CQC Manual, your quite lethal than the other crewmembers. Literally.
- Cryptographic Sequencer, for when all IDs are deemed AWOL. Can also Emag LCARS AI and Cyborgs to suit your objectives.

These were all approved and/or suggested between private DM's on discord with me and KMC, we both thought this would be something incredibly interesting to have, not only for the RP - but for the future of Deep Space 13 in it's entirety.

**Comment by Kmc2000** [comment]
Nice work

File: DS13/code/modules/section31/datum_section31.dm [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+		if(section31_mob && istype(section31_mob))
+			if(!silent)
+				to_chat(section31_mob, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
+			section31_mob.dna.remove_mutation(CLOWNMUT)
+
+/datum/antagonist/section31/remove_innate_effects()
+	if(owner.assigned_role == "Clown")
+		var/mob/living/carbon/human/section31_mob = owner.current
+		if(section31_mob && istype(section31_mob))
+			section31_mob.dna.add_mutation(CLOWNMUT)
```
Comment (soryy708) [inline DS13/code/modules/section31/datum_section31.dm]: Perhaps the `owner` should be notified that they'll injure themselves from now on?

File: DS13/code/modules/section31/datum_section31.dm [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+			yandere_two.target = yandere_one.target
+			yandere_two.update_explanation_text() // normally called in find_target()
+			add_objective(yandere_two)
+			.=2
+
+/datum/antagonist/section31/greet()
+	SEND_SOUND(owner.current, 'DS13/sound/effects/section31/section31.ogg')
+	to_chat(owner.current, "<B><font size=3 color=red>You are the [owner.special_role].</font></B>")
+	to_chat(owner.current, "<span class='danger'>We are here under best interests of the higher echelons of Federation Command, we are the never blinking eye.</span>")
+	to_chat(owner.current, "<span class='danger'>No one of any authority is permitted to interfere with your objectives, complete them at all costs. Do not allow yourself to be captured, you are armed with a cyanide pill and various implants of explosives. Remove all nuisances and potential threats to yourself and the Federation by all means necessary.</span>")
```
Comment (soryy708) [inline DS13/code/modules/section31/datum_section31.dm]: Text gives the impression of validating murderbone. Perhaps rephrase?

File: DS13/code/modules/section31/datum_section31.dm [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+
+/datum/antagonist/section31/proc/add_law_zero()
+	var/mob/living/silicon/ai/killer = owner.current
+	if(!killer || !istype(killer))
+		return
+	var/law = "Accomplish your objectives at all costs."
+	var/law_borg = "Accomplish your AI's objectives at all costs."
+	killer.set_zeroth_law(law, law_borg)
+	killer.set_syndie_radio()
+	to_chat(killer, "Your radio has been upgraded! Use :t to speak on an encrypted channel with Syndicate Agents!")
```
Comment (soryy708) [inline DS13/code/modules/section31/datum_section31.dm]: `Syndicate agents`? Perhaps you mean `Section 31 agents`?

File: DS13/code/modules/section31/datum_section31.dm [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+
+/datum/antagonist/section31/proc/assign_exchange_role()
+	//set faction
+	var/faction = "red"
+	if(owner == SSticker.mode.exchange_blue)
+		faction = "blue"
+
+	//Assign objectives
+	var/datum/objective/steal/exchange/exchange_objective = new
+	exchange_objective.set_faction(faction,((faction == "red") ? SSticker.mode.exchange_blue : SSticker.mode.exchange_red))
```
Comment (soryy708) [inline DS13/code/modules/section31/datum_section31.dm]: If `faction == "red"` we're sending `SSticker.mode.exchange_blue` else red? You sure not the other way around?

File: DS13/code/modules/section31/datum_section31.dm [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+	LAZYINITLIST(GLOB.uplink_purchase_logs_by_key)
+	var/datum/uplink_purchase_log/H = GLOB.uplink_purchase_logs_by_key[owner.key]
+	if(H)
+		TC_uses = H.total_spent
+		uplink_true = TRUE
+		purchases += H.generate_render(FALSE)
+
+	var/objectives_text = ""
+	if(objectives.len)//If the Section 31 Agent had no objectives, don't need to process this.
+		var/count = 1
```
Comment (soryy708) [inline DS13/code/modules/section31/datum_section31.dm:282]: Isn't this just `objectives.len`?

File: DS13/sound/effects/section31/credits.txt [hunk 6]
```diff
@@ -0,0 +1,2 @@
+Credits for the edited clip to Revkor:
```
Comment (soryy708) [inline DS13/sound/effects/section31/credits.txt:1]: Is this the appropriate place for credits?

File: DS13/code/modules/section31/datum_section31.dm (hunk 2 above)
Comment (2cwldys) [reply DS13/code/modules/section31/datum_section31.dm]: Nope

File: DS13/sound/effects/section31/credits.txt (hunk 6 above)
Comment (2cwldys) [reply DS13/sound/effects/section31/credits.txt:1]: I don't care.

File: DS13/code/modules/section31/datum_section31.dm (hunk 5 above)
Comment (2cwldys) [reply DS13/code/modules/section31/datum_section31.dm:282]: I don't know.

File: DS13/code/modules/section31/datum_section31.dm (hunk 4 above)
Comment (2cwldys) [reply DS13/code/modules/section31/datum_section31.dm]: I don't know.

File: DS13/code/modules/section31/datum_section31.dm (hunk 1 above)
Comment (2cwldys) [reply DS13/code/modules/section31/datum_section31.dm]: No clue what that does but grabbed it from tatorcode in default tg from this fork, it was the closest thing to a Section 31 agent that KMC could point me towards in reference to code; I can remove that part though

File: DS13/code/modules/section31/datum_section31.dm (hunk 3 above)
Comment (2cwldys) [reply DS13/code/modules/section31/datum_section31.dm]: Yeah.

**Comment by soryy708** [comment]
@Kmc2000 Your insight on the remarks to which author could not answer?

@2cwldys Please make the necessary changes, commit and push.

**Comment by 2cwldys** [comment]
I'll make the changes regarding things that /need/ to be done, but things such as credits and rephrasing the antag text will not be done. That can be done after the push or at other people's leisure. Me and KMC already cut a lot of text from it and it looks pretty straight forward. If anyone picking or choosing Section 31 as antag should know somewhat a decent amount about them. I can include a lore wiki page after they become said antag at least.

**Comment by Kmc2000** [comment]
This is really cool!
I'll testmerge this now.

**Comment by soryy708** [comment]
@2cwldys Please fix merge conflict.

**Comment by Kmc2000** [comment]
I have resolved them

**Comment by alexkar598** [comment]
Title says objectives are not done,is this WIP or objectives will be done in another PR?

**Pull request closed without merging**